torch/optim/optimizer.py

```diff
@@ -790,6 +790,24 @@ def update_group(group: Dict[str, Any], new_group: Dict[str, Any]) -> Dict[str,
             post_hook(self)
 
 
+    def init_state_per_param(self, param: torch.Tensor, param_group: Dict[str, Any]) -> None:
```

Comment:
Should we explicitly document for each optimizer when they support this feature and raise an error if it is overwritten on an optimizer that doesn't support it?